front/app/containers/Admin/projects/project/phase/phaseParticipationConfig/index.tsx

```diff
@@ -115,591 +65,490 @@ const MAX_VOTES_PER_VOTING_METHOD: Record<VotingMethod, number> = {
   budgeting: 100,
 };
 
-class PhaseParticipationConfig extends PureComponent<
-  Props & WrappedComponentProps,
-  State
-> {
-  subscriptions: Subscription[];
-
-  constructor(props: Props & WrappedComponentProps) {
-    super(props);
-
-    this.state = {
-      participation_method: 'ideation',
-      posting_enabled: true,
-      commenting_enabled: true,
-      reacting_enabled: true,
-      reacting_like_method: 'unlimited',
-      reacting_like_limited_max: null,
-      reacting_dislike_enabled: true,
-      allow_anonymous_participation: false,
-      voting_method: null,
-      reacting_dislike_method: 'unlimited',
-      reacting_dislike_limited_max: null,
-      presentation_mode: 'card',
-      voting_min_total: null,
-      voting_max_total: null,
-      survey_service: null,
-      survey_embed_url: null,
-      voting_max_votes_per_idea: null,
-      loaded: false,
-      noLikingLimitError: null,
-      noDislikingLimitError: null,
-      minTotalVotesError: null,
-      maxTotalVotesError: null,
-      maxVotesPerOptionError: null,
-      voting_term_plural_multiloc: null,
-      voting_term_singular_multiloc: null,
-      voteTermError: null,
-      poll_anonymous: false,
-      ideas_order: 'trending',
-      input_term: 'idea',
-      document_annotation_embed_url: null,
-      appConfig: props.appConfig || null,
+const PhaseParticipationConfig = ({
+  phase,
+  onSubmit,
+  onChange,
+  className,
+  apiErrors,
+}: Props) => {
+  const surveys_enabled = useFeatureFlag({ name: 'surveys' });
+  const typeform_enabled = useFeatureFlag({ name: 'typeform_surveys' });
+  const google_forms_enabled = useFeatureFlag({ name: 'google_forms_surveys' });
+  const survey_monkey_enabled = useFeatureFlag({
+    name: 'surveymonkey_surveys',
+  });
+  const enalyzer_enabled = useFeatureFlag({ name: 'enalyzer_surveys' });
+  const survey_xact_enabled = useFeatureFlag({ name: 'survey_xact_surveys' });
+  const qualtrics_enabled = useFeatureFlag({ name: 'qualtrics_surveys' });
+  const smartsurvey_enabled = useFeatureFlag({ name: 'smart_survey_surveys' });
+  const snap_survey_enabled = useFeatureFlag({ name: 'snap_survey_surveys' });
+  const microsoft_forms_enabled = useFeatureFlag({
+    name: 'microsoft_forms_surveys',
+  });
+
+  const { data: appConfig } = useAppConfiguration();
+  const { formatMessage } = useIntl();
+  const [participationConfig, setParticipationConfig] =
```

Comment:
Decided to go with a single state here, instead of keeping one for each attribute.